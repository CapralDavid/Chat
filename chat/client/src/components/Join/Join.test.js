//'use strict';  //надо ли?

//import React from 'react';

//import renderer from 'react-test-renderer';
import { create, act } from "react-test-renderer";

import Join from './Join.js';


const three = 3;

test('idi naxui', () => {
    expect(three).toEqual(3);

}); 

//проверяем окно выбора имени и комнаты 
//test('');


//создаем join и потом сравниваем его с снапшотом
// describe('test', () => {
//     test('test1', () =>{
//         let component;
//         act(() =>{
//             //component = create(<Join> </Join>)
//         })
//     })
// });


/*

it('renders correctly', () => {
  const tree = renderer
    .create(<Link page="http://www.facebook.com">Facebook</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('should handle the click event', () => {
  window.alert = jest.fn();
  const output = shallow(
    <Link title="mockTitle" url="mockUrl" />
  );
  output.simulate('click');
  expect(window.alert).toHaveBeenCalledWith('Кликнули по ссылке!');
});




//сравниваем элемент с снапшотом

// render the component
let root; 
act(() => {
  root = create(<App value={1}/>)
});

// make assertions on root 
expect(root.toJSON()).toMatchSnapshot();

// update with some different props
act(() => {
  root.update(<App value={2}/>);
})

// make assertions on root 
expect(root.toJSON()).toMatchSnapshot();



*/ 