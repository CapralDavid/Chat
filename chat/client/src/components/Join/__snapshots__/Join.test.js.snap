// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render Join element render correct Join 1`] = `
<div
  className="joinOuterContainer"
>
  <div
    className="joinInnerContainer"
  >
    <h1
      className="heading"
    >
      Join Chat
    </h1>
    <div>
      <input
        className="joinInput"
        onChange={[Function]}
        placeholder="Name"
        type="text"
      />
    </div>
    <div>
      <input
        className="joinInput mt-20"
        onChange={[Function]}
        placeholder="Room"
        type="text"
      />
    </div>
    <a
      href="/chat?name=&room="
      onClick={[Function]}
    >
      <button
        className="button mt-20"
        type="submit"
      >
        Enter
      </button>
    </a>
  </div>
</div>
`;

exports[`test render Join element render correct: Join with argument  1`] = `
<div
  className="joinOuterContainer"
>
  <div
    className="joinInnerContainer"
  >
    <h1
      className="heading"
    >
      Join Chat
    </h1>
    <div>
      <input
        className="joinInput"
        onChange={[Function]}
        placeholder="Name"
        type="text"
      />
    </div>
    <div>
      <input
        className="joinInput mt-20"
        onChange={[Function]}
        placeholder="Room"
        type="text"
      />
    </div>
    <a
      href="/chat?name=&room="
      onClick={[Function]}
    >
      <button
        className="button mt-20"
        type="submit"
      >
        Enter
      </button>
    </a>
  </div>
</div>
`;
